### PR TITLE
Add kafka support to CLI (fbsql)

### PR DESCRIPTION
**Pull request by travisturner**
This commit adds the ability to provide a `--kafka-config` command line argument referncing a toml file to configure kafka.

### CODE REVIEW

File: .gitlab/.gitlab-ci.yml [hunk 1]
```diff
@@ -636,14 +668,16 @@ s3 dump:
     - |
       for goos in "darwin" "linux"; do
         for goarch in "amd64" "arm64"; do
-          for binary in "featurebase" "fbsql"; do
-            aws s3 cp ${binary}_${goos}_${goarch} s3://molecula-artifact-storage/featurebase/${CI_COMMIT_BRANCH}/${CI_COMMIT_SHORT_SHA}/${binary}_${goos}_${goarch}
-            aws s3 cp ${binary}_${goos}_${goarch} s3://molecula-artifact-storage/featurebase/${CI_COMMIT_BRANCH}/_latest/${binary}_${goos}_${goarch}
-          done
+          aws s3 cp featurebase_${goos}_${goarch} s3://molecula-artifact-storage/featurebase/${ci_commit_branch}/${ci_commit_short_sha}/featurebase_${goos}_${goarch}
```
Comment (gardavi) [inline .gitlab/.gitlab-ci.yml:671]: `${ci_commit_branch}` and `${ci_commit_short_sha}` need to be capitalized or they end up `""` (empty string)

**Review by gardavi — DISMISSED** [review]

File: .gitlab/.gitlab-ci.yml (hunk 1 above)
Comment (travisturner) [reply .gitlab/.gitlab-ci.yml]: Crap. Sometimes I accidentally type `u` in vim while I have text highlighted and it lowercases everything. I must have done that here without realizing it.

Thanks for catching that.

**Comment by JasonAllison** [comment]
@travisturner - is this ready for argon team to review?

**Comment by travisturner** [comment]
> @travisturner - is this ready for argon team to review?

@JasonAllison yes

**Comment by tgruben** [comment]
The ability to provide additional confluent (librdkafka) configuration in a json file demonstrated in the idk kafka consumer source `KafkaConfiguration` has proven very useful in both tuning and debugging kafka issues from the clients perspective.  It allows access to all these parameters https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md.  You might consider adding in that ability.

**Comment by travisturner** [comment]
I agree with you @tgruben. I think the functionality is there, it's just not currently exposed. The original specs for this work asked me to make a 1-to-1 mapping of JSON message attributes to FeatureBase columns, with no ability to customize anything. I thought that was a little too restrictive, so what I wrote is a compromise; I exposed some ability to configure things, but not everything.

File: idk/ingest.go [hunk 2]
```diff
@@ -240,7 +244,17 @@ func (m *Main) Rename() {
 	}
 }
 
+// Basic sets up Main with basic functionality, excluding those things which are
+// not required for some implementations (such as the kafka runner in fbsql).
+func (m *Main) Basic() {
```
Comment (jaffee) [inline idk/ingest.go:249]: rename to SetBasic? Just Basic makes it sound like a "getter"

Comment (travisturner) [reply idk/ingest.go]: good call. I changed that and rebased everything.

**Review by jaffee — APPROVED** [review]

**Pull request merged**